tools/nightly_build_and_test_stats.groovy

```diff
@@ -56,35 +51,50 @@ def isGaTag(String version, String tag) {
     }
 }
 
-// Get the latest upstream openjdk build tag
-def getLatestOpenjdkBuildTag(String version) {
-    def openjdkRepo = "https://github.com/openjdk/${version}.git"
-    if (version == "aarch32-jdk8u") {
+// Determine the upstream OpenJDK reporistory
+def getUpstreamRepo(String version) {
+    def openjdkRepo
+
+    def versionInt = (version == "aarch32-jdk8u" || version == "alpine-jdk8u") ? 8 : version.replaceAll("u", "").replaceAll("jdk", "").toInteger()
+    def isUpdateVersion = version.endsWith("u")
+    
+    if (versionInt >= 23 && !isUpdateVersion) {
+        // jdk-23+ stabalisation versions are now branches in jdk(head) repo
+        openjdkRepo = "https://github.com/openjdk/jdk.git"
+    } else if (version == "aarch32-jdk8u") {
         openjdkRepo = "https://github.com/openjdk/aarch32-port-jdk8u.git"
-    } else if (version == "alpine-jdk8u") {
+    } else if (version.contains("jdk8u")) {
         openjdkRepo = "https://github.com/openjdk/jdk8u.git"
+    } else {
+        openjdkRepo = "https://github.com/openjdk/${version}.git"
     }
 
-    // Need to include jdk8u to avoid picking up old tag format    
-    def jdk8Filter = (version.contains("jdk8u")) ? "| grep 'jdk8u'" : ""
+    return openjdkRepo
+}
+
+// Get the latest upstream openjdk build tag
+def getLatestOpenjdkBuildTag(String version) {
+    def openjdkRepo = getUpstreamRepo(version)
+    def versionInt = (version == "aarch32-jdk8u" || version == "alpine-jdk8u") ? 8 : version.replaceAll("u", "").replaceAll("jdk", "").toInteger()
```

Comment:
Could these two `replaceAll` invocations  be replaced with the simpler `replaceAll("[a-z]","");` to strip out all letters from the name?

Reply:
I think so